src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,223 @@
+            while (resultSet.next()) {
+                Manufacturer manufacturerEntity = createManufacturerEntity(resultSet);
+                Car carEntity = createCarEntity(resultSet, manufacturerEntity);
+                cars.add(carEntity);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+        List<Driver> driversForCar;
+        for (Car car : cars) {
+            driversForCar = getDriversForCar(car.getId());
+            car.setDrivers(driversForCar);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarRequest = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement =
+                        connection.prepareStatement(updateCarRequest)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setObject(2, car.getManufacturer().getId());
+            updateCarStatement.setObject(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a car " + car, e);
+        }
+        removeCarDrivers(car);
+        insertCarDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarRequest = "UPDATE cars SET is_deleted = TRUE WHERE id = ?"
+                + "AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement =
+                        connection.prepareStatement(deleteCarRequest)) {
+            deleteCarStatement.setObject(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete a car with ID " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCarsQuery = "SELECT id, model, manufacturer_id FROM cars AS c "
+                + "INNER JOIN cars_drivers AS cd "
+                + "ON c.id = cd.car_id "
+                + "WHERE is_deleted = FALSE AND driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversStatement =
+                        connection.prepareStatement(getCarsQuery)) {
+            getDriversStatement.setObject(1, driverId);
+            ResultSet resultSet = getDriversStatement.executeQuery();
+            while (resultSet.next()) {
+                Long carId = resultSet.getObject("id", Long.class);
+                get(carId).ifPresent(cars::add);
```

Comment:
you should write a query that will help you get cars directly without using additional get method